backends-velox/src-delta/main/scala/org/apache/gluten/component/VeloxDeltaComponent.scala

```
@@ -25,12 +25,24 @@ import org.apache.gluten.extension.columnar.validator.Validators
 import org.apache.gluten.extension.injector.Injector
 
 import org.apache.spark.sql.execution.{FileSourceScanExec, FilterExec, ProjectExec}
+import org.apache.spark.util.SparkReflectionUtil
 
 class VeloxDeltaComponent extends Component {
   override def name(): String = "velox-delta"
   override def buildInfo(): Component.BuildInfo =
     Component.BuildInfo("VeloxDelta", "N/A", "N/A", "N/A")
   override def dependencies(): Seq[Class[_ <: Component]] = classOf[VeloxBackend] :: Nil
+
+  override def isRuntimeCompatible: Boolean = {
+    try {
+      SparkReflectionUtil.classForName("io.delta.sql.DeltaSparkSessionExtension")
+      true
+    } catch {
+      case _: ClassNotFoundException =>
+        false
```

Should we fail fast or add some warn logs?